python-package/lightgbm/sklearn.py

```diff
@@ -530,6 +531,8 @@ def predict(self, X, raw_score=False, num_iteration=-1,
                              "match the input. Model n_features_ is %s and "
                              "input n_features is %s "
                              % (self._n_features, n_features))
+        if num_iteration is None:
+            num_iteration = self.best_iteration_ if self.best_iteration_ is not None else -1
```

Comment:
Is this just `num_iteration = self.best_iteration_`  because `self.best_iteration` is defult -1?
https://github.com/Microsoft/LightGBM/pull/1575/files#r209197852

Reply:
Good catch!

It seems that we even can remove this line completely, because of
https://github.com/Microsoft/LightGBM/blob/c78f26ed9e20575ab770d0dfa2800d9438c0f4c6/python-package/lightgbm/sklearn.py#L275
https://github.com/Microsoft/LightGBM/blob/c78f26ed9e20575ab770d0dfa2800d9438c0f4c6/python-package/lightgbm/sklearn.py#L486
https://github.com/Microsoft/LightGBM/blob/c78f26ed9e20575ab770d0dfa2800d9438c0f4c6/python-package/lightgbm/sklearn.py#L581-L585

Reply:
Ah, you're right!
we don't need this change.